main.js

```diff
@@ -0,0 +1,362 @@
+// Javascript 
+
+// array variable to contain all posts
+var posts = [];

```

Comment:
Would recommend using let/const to declare variables since you are using Class which was added in ES6